pkg/controller/gc.go

```diff
@@ -444,6 +471,9 @@ func (c *Controller) gcLoadBalancer() error {
 
 	vpcLbs := strset.NewWithSize(len(dnats))
 	for _, dnat := range dnats {
+		if c.matchOmitName(dnat.Name) {
+			continue
+		}
```

Comment:
I think we may want to add ext-id check here as some services may use the name field with UUID's so we'd need to check other fields to match our omit values against.  
`if c.matchOmitExternalIDs(dnat.ExternalIDs) {
			continue
}`

Reply:
Sadly `dnat` doesn't have a defined value for `ExternalIDs`, I'll have to tinker with this to see what I can figure out.

Reply:
I ended up adding the omit filters to the closure for the lbs, which resolves the issue where GC would eliminate out of band OVN loadbalancers.

* https://github.com/kubeovn/kube-ovn/pull/4997/files#diff-efdb6355f2187b31e5e61610c24bf4d446a23d814d51448c170a91ef96ae1e89R666